tests/regression_opcodes.rs

```diff
@@ -0,0 +1,276 @@
+    regression!(OP_RETURN_197, "OP_RETURN_197");
+    regression!(OP_RETURN_198, "OP_RETURN_198");
+    regression!(OP_RETURN_199, "OP_RETURN_199");
+    regression!(OP_RETURN_200, "OP_RETURN_200");
+    regression!(OP_RETURN_201, "OP_RETURN_201");
+    regression!(OP_RETURN_202, "OP_RETURN_202");
+    regression!(OP_RETURN_203, "OP_RETURN_203");
+    regression!(OP_RETURN_204, "OP_RETURN_204");
+    regression!(OP_RETURN_205, "OP_RETURN_205");
+    regression!(OP_RETURN_206, "OP_RETURN_206");
+    regression!(OP_RETURN_207, "OP_RETURN_207");
+    regression!(OP_RETURN_208, "OP_RETURN_208");
+    regression!(OP_RETURN_209, "OP_RETURN_209");
+    regression!(OP_RETURN_210, "OP_RETURN_210");
+    regression!(OP_RETURN_211, "OP_RETURN_211");
+    regression!(OP_RETURN_212, "OP_RETURN_212");
+    regression!(OP_RETURN_213, "OP_RETURN_213");
+    regression!(OP_RETURN_214, "OP_RETURN_214");
+    regression!(OP_RETURN_215, "OP_RETURN_215");
+    regression!(OP_RETURN_216, "OP_RETURN_216");
+    regression!(OP_RETURN_217, "OP_RETURN_217");
+    regression!(OP_RETURN_218, "OP_RETURN_218");
+    regression!(OP_RETURN_219, "OP_RETURN_219");
+    regression!(OP_RETURN_220, "OP_RETURN_220");
+    regression!(OP_RETURN_221, "OP_RETURN_221");
+    regression!(OP_RETURN_222, "OP_RETURN_222");
+    regression!(OP_RETURN_223, "OP_RETURN_223");
+    regression!(OP_RETURN_224, "OP_RETURN_224");
+    regression!(OP_RETURN_225, "OP_RETURN_225");
+    regression!(OP_RETURN_226, "OP_RETURN_226");
+    regression!(OP_RETURN_227, "OP_RETURN_227");
+    regression!(OP_RETURN_228, "OP_RETURN_228");
+    regression!(OP_RETURN_229, "OP_RETURN_229");
+    regression!(OP_RETURN_230, "OP_RETURN_230");
+    regression!(OP_RETURN_231, "OP_RETURN_231");
+    regression!(OP_RETURN_232, "OP_RETURN_232");
+    regression!(OP_RETURN_233, "OP_RETURN_233");
+    regression!(OP_RETURN_234, "OP_RETURN_234");
+    regression!(OP_RETURN_235, "OP_RETURN_235");
+    regression!(OP_RETURN_236, "OP_RETURN_236");
+    regression!(OP_RETURN_237, "OP_RETURN_237");
+    regression!(OP_RETURN_238, "OP_RETURN_238");
+    regression!(OP_RETURN_239, "OP_RETURN_239");
+    regression!(OP_RETURN_240, "OP_RETURN_240");
+    regression!(OP_RETURN_241, "OP_RETURN_241");
+    regression!(OP_RETURN_242, "OP_RETURN_242");
+    regression!(OP_RETURN_243, "OP_RETURN_243");
+    regression!(OP_RETURN_244, "OP_RETURN_244");
+    regression!(OP_RETURN_245, "OP_RETURN_245");
+    regression!(OP_RETURN_246, "OP_RETURN_246");
+    regression!(OP_RETURN_247, "OP_RETURN_247");
+    regression!(OP_RETURN_248, "OP_RETURN_248");
+    regression!(OP_RETURN_249, "OP_RETURN_249");
+    regression!(OP_RETURN_250, "OP_RETURN_250");
+    regression!(OP_RETURN_251, "OP_RETURN_251");
+    regression!(OP_RETURN_252, "OP_RETURN_252");
+    regression!(OP_RETURN_253, "OP_RETURN_253");
+    regression!(OP_RETURN_254, "OP_RETURN_254");
+    regression!(OP_INVALIDOPCODE, "OP_INVALIDOPCODE");
+}
```

Comment:
First of all, this comment is not very important, especially compared to the others.

Just noticed that every single output here is just stringification of the input. You could write this macro instead:

```rust
macro_rules! test_opcodes {
    ($($op:ident),* $(,)?) => {
        $(
            let op = bitcoin::blockdata::opcodes::all::$op;
            let want = concat!("\"", strigify!($op), "\"");
            let got = ::serde_json::to_string(&op).unwrap();
            assert_eq!(got, want, "Regression test failed for: {}", stringify!($op));
        )*
    }
}

test_opcodes! {
    OP_PUSHBYTES_0,
    OP_PUSHBYTES_1,
    // ...
}
```

Notes:
* I prefer to use macros with repeated inputs if I call them like this because it avoids useless repeating of macro name.
* Notice `stringify!()` macro
* I used `concat!()` instead of `format!()` because it's compile-time and hopefully runs the test faster which may be important, but TBH didn't measure. Could also make it slower if it's recompiled often and the compiler does more work because of it.
* Didn't use `concat!()` in `assert_eq` because it's in unlikely branch and format string is a bit more readable
* `Regression test failed for` message may be unneeded as one should see the opcode name in the `assert_eq` output ("right: ...")

Reply:
Thanks for taking the time to explain this. Re last point; I checked, error is clear without the message.